platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbDataSource.java

```diff
@@ -1124,23 +1126,30 @@ private void writeHashes(final long maxValidPath, final Stream<VirtualHashRecord
     private void writeLeavesToPathToKeyValue(
             final long firstLeafPath,
             final long lastLeafPath,
-            final Stream<VirtualLeafRecord<K, V>> dirtyLeaves,
-            final Stream<VirtualLeafRecord<K, V>> deletedLeaves,
+            @NonNull final Stream<VirtualLeafRecord<K, V>> dirtyLeaves,
+            @NonNull final Stream<VirtualLeafRecord<K, V>> deletedLeaves,
             boolean isReconnect)
             throws IOException {
-        if ((dirtyLeaves == null) || (firstLeafPath <= 0)) {
-            // nothing to do
+        // If both streams are empty, no new data files should be created. One simple way to
+        // check emptiness is to use iterators. The streams aren't processed in parallel anyway
+        final Iterator<VirtualLeafRecord<K, V>> dirtyIterator = dirtyLeaves
+                .sorted(Comparator.comparingLong(VirtualLeafRecord::getPath))
+                .iterator();
+        final Iterator<VirtualLeafRecord<K, V>> deletedIterator = deletedLeaves.iterator();
+
+        if (!dirtyIterator.hasNext() && !deletedIterator.hasNext()) {
+            // Nothing to do
```

Comment:
It's odd that GitHub shows these changes here as they belong to a separate PR/commit ( https://github.com/hashgraph/hedera-services/pull/14020 ). Looks like a bug in GitHub.

You may want to pull latest changes from develop into your branch and update your PR to try and fix it. This should hopefully remove `platform-data` from the list of reviewers.

Reply:
Hmm that was confusing, hopefully good now. 🤔

Reply:
Thanks for fixing this! There's no longer any files owned by platform-data, so you don't need our review on your PR anymore.